mailet/standard/src/main/java/org/apache/james/transport/matchers/dlp/DlpRulesLoader.java

```diff
@@ -47,19 +53,36 @@ public DlpDomainRules load(Domain domain) {
         }
 
         private DlpDomainRules toRules(Stream<DLPConfigurationItem> items) {
-            DlpDomainRules.DlpDomainRulesBuilder builder = DlpDomainRules.builder();
-            items.forEach(item -> {
-                if (item.getTargets().isContentTargeted()) {
-                    builder.contentRule(item.getId(), Pattern.compile(item.getRegexp()));
-                }
-                if (item.getTargets().isRecipientTargeted()) {
-                    builder.recipientRule(item.getId(), Pattern.compile(item.getRegexp()));
-                }
-                if (item.getTargets().isSenderTargeted()) {
-                    builder.senderRule(item.getId(), Pattern.compile(item.getRegexp()));
-                }
-            });
-            return builder.build();
+            return DlpDomainRules.of(items.flatMap(this::toRules)
+                .collect(Guavate.toImmutableList()));
+        }
+
+        private Stream<DlpDomainRule> toRules(DLPConfigurationItem item) {
+            return StreamUtils.flatten(
+                toRecipientsRule(item),
+                toSenderRule(item),
+                toContentRule(item));
+        }
+
+        private Stream<DlpDomainRule> toRecipientsRule(DLPConfigurationItem item) {
+            return toRule(item, factory::recipientRule);
+        }
+
+        private Stream<DlpDomainRule> toSenderRule(DLPConfigurationItem item) {
+            return toRule(item, factory::senderRule);
+        }
+
+        private Stream<DlpDomainRule> toContentRule(DLPConfigurationItem item) {
+            return toRule(item, factory::contentRule);
         }
+
+        private Stream<DlpDomainRule> toRule(DLPConfigurationItem item,
+                                                   BiFunction<DLPConfigurationItem.Id, Pattern, DlpDomainRule> toRule) {
+            if (item.getTargets().isRecipientTargeted()) {
```

Comment:
you handle only recipient target or am I missing something ?

Reply:
Oups, looks like a stupid issue. Sory.